### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by jesusbv**

### CODE REVIEW

**Comment by jesusbv** [comment]
version `0.0.5` is on #10

File: lib/instance_billing_flavor_check/utils.py [hunk 1]
```diff
@@ -139,6 +141,42 @@ def get_rmt_ip_addr():
...
+    """
+    Get proxy info.
+
+    If any proxy has been set up,
+    this method returns the proxy info in a dictionary.
+
+    Otherwise, returns None
+    """
+    if os.environ.get('http_proxy') or os.environ.get('https_proxy'):
+        logger.info('Using proxy settings present on the environment.')
```
Comment (rjschwei) [inline lib/instance_billing_flavor_check/utils.py:154]: s/on/in/

File: lib/instance_billing_flavor_check/utils.py [hunk 2]
```diff
@@ -139,6 +141,42 @@ def get_rmt_ip_addr():
...
+        # HTTP_PROXY and HTTPS_PROXY
+        return None
+
+    proxies = {}
+    proxy_config = []
+    try:
+        with open(PROXY_CONFIG_PATH, 'r') as proxy_file:
+            proxy_config = proxy_file.readlines()
+    except FileNotFoundError:
+        logging.info('No proxy config file found on %s', PROXY_CONFIG_PATH)
```
Comment (rjschwei) [inline lib/instance_billing_flavor_check/utils.py:165]: The user is not telling us to use a proxy, it is a discovery process, as such there is no need to tell the user anything about our discovery process. `pass` or `return`

File: lib/instance_billing_flavor_check/utils.py [hunk 3]
```diff
@@ -139,6 +141,42 @@ def get_rmt_ip_addr():
...
+        if 'PROXY_ENABLED' in entry and 'no' in entry:
+            return None
+        if 'HTTP_PROXY' in entry:
+            proxies['http_proxy'] = entry.split('"')[1]
+        if 'HTTPS_PROXY' in entry:
+            proxies['https_proxy'] = entry.split('"')[1]
+        if 'NO_PROXY' in entry:
+            proxies['no_proxy'] = entry.split('"')[1]
+
+    return proxies or None
```
Comment (rjschwei) [inline lib/instance_billing_flavor_check/utils.py:177]: The empty dict will evaluate to `False` anywhere we have to check if there is a proxy as such the return of `None` is superfluous.

File: lib/instance_billing_flavor_check/utils.py [hunk 4]
```diff
@@ -147,12 +185,14 @@ def make_request(rmt_ip_addr, metadata, identifier):
...
     }
+    proxies = _get_proxies()
     try:
         response = requests.get(
             instance_check_url,
             timeout=2,
             verify=False,
-            params=billing_check_params
+            params=billing_check_params,
+            proxies=proxies
```
Comment (rjschwei) [inline lib/instance_billing_flavor_check/utils.py:195]: Will work with an empty `{}`

**Review by rjschwei — CHANGES_REQUESTED** [review]

**Review by rjschwei — APPROVED** [review]

**Pull request merged**